Makefile

```diff
@@ -4,8 +4,19 @@ ROM ?= 256
 
 # Dump memory information: on = print allocs, full = print allocs + dump pools
 TRACE ?= off
+
 # Generate and run snapshot as byte code instead of running JS directly
+ifneq (,$(filter $(MAKECMDGOALS),ide ashell linux))
+SNAPSHOT=off
+# if the user passes in SNAPSHOT=on for ide, ashell, or linux give an error
+ifeq ($(SNAPSHOT), on)
+$(error ide, ashell, and linux do not support SNAPSHOT=$(SNAPSHOT))
+endif
+else
+# snapshot is enabled by default
 SNAPSHOT ?= on
+EXT_JERRY_FLAGS += -DFEATURE_JS_PARSER=OFF
```

Comment:
Putting this here looks like a bug. if the user set it to SNAPSHOT=off, it will be off even though you attempted to turn it on here. So you still need to tests actual SNAPSHOT value as before.